File: Count Elements With Maximum Frequency/Count Elements With Maximum Frequency.cpp
// Solution for Count Elements With Maximum Frequency in CPP

class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        // Frequency array for numbers in range [1..100]
        vector<int> frequency(101, 0);

        int maxFrequency = 0;   // Track highest frequency of any element
        int totalCount = 0;     // Track sum of frequencies of all elements with max frequency

        for (int num : nums) {
            // Increase count for this number
            frequency[num]++;

            // If this number becomes the new max frequency
            if (frequency[num] > maxFrequency) {
                maxFrequency = frequency[num];     // Update highest frequency
                totalCount = maxFrequency;         // Reset totalCount to this frequency
            }
            // If this number just matched the current max frequency
            else if (frequency[num] == maxFrequency) {
                totalCount += maxFrequency;        // Add its contribution
            }
        }

        return totalCount;
    }
};

File: Count Elements With Maximum Frequency/Count Elements With Maximum Frequency.py
# Solution for Count Elements With Maximum Frequency in PY

class Solution:
    def maxFrequencyElements(self, nums: list[int]) -> int:
        freq = [0] * 101
        maxFreq = total = 0

        for num in nums:
            freq[num] += 1
            if freq[num] > maxFreq:
                maxFreq = freq[num]
                total = maxFreq
            elif freq[num] == maxFreq:
                total += maxFreq
        return total
